### DISCUSSION TITLE
Isso doesn't want to start

### DISCUSSION THREAD

**Discussion by inputuserhere** [posativ/isso#776] [Q&A]
I can't figure out what's exactly wrong. Also, how do I see the Isso version?

isso@vm-150:~$ sudo systemctl status isso

 ```
 isso.service - Isso Commenting Server
   Loaded: loaded (/etc/systemd/system/isso.service; enabled; vendor preset: enab
   Active: activating (auto-restart) (Result: exit-code) since Tue 2020-03-24 19:
  Process: 9223 ExecStart=/home/isso/.local/bin/isso -c /home/isso/isso.conf (cod
 Main PID: 9223 (code=exited, status=2)

Mar 24 19:13:12 vm-150 systemd[1]: isso.service: Unit entered failed state.
Mar 24 19:13:12 vm-150 systemd[1]: isso.service: Failed with result 'exit-code'
lines 1-8/8 (END)
```

**Comment by Fryboyter** [comment]
Do you get more details about what goes wrong when you run "sudo journalctl -u isso" and / or "sudo journalctl -xe"? If yes which? Please also publish the content of the file isso.service. The file /home/isso/isso.conf actually exists and it has the correct file permissions?

**Comment by inputuserhere** [comment]
> sudo journalctl -u isso


```
-- Logs begin at Thu 2020-03-26 11:51:33 EDT, end at Fri 2020-03-27 00:21:59 EDT
Mar 26 11:51:33 vm-150 systemd[1]: Started Isso Commenting Server.
Mar 26 11:51:33 vm-150 isso[26478]: usage: isso [-h] [--version] [-c /etc/isso
Mar 26 11:51:33 vm-150 isso[26478]: isso: error: too few arguments
Mar 26 11:51:33 vm-150 systemd[1]: isso.service: Main process exited, code=exi
Mar 26 11:51:33 vm-150 systemd[1]: isso.service: Unit entered failed state.
Mar 26 11:51:33 vm-150 systemd[1]: isso.service: Failed with result 'exit-code
Mar 26 11:51:38 vm-150 systemd[1]: isso.service: Service hold-off time over, s
Mar 26 11:51:38 vm-150 systemd[1]: Stopped Isso Commenting Server.
Mar 26 11:51:38 vm-150 systemd[1]: Started Isso Commenting Server.
Mar 26 11:51:39 vm-150 isso[26480]: usage: isso [-h] [--version] [-c /etc/isso
Mar 26 11:51:39 vm-150 isso[26480]: isso: error: too few arguments
Mar 26 11:51:39 vm-150 systemd[1]: isso.service: Main process exited, code=exi
Mar 26 11:51:39 vm-150 systemd[1]: isso.service: Unit entered failed state.
Mar 26 11:51:39 vm-150 systemd[1]: isso.service: Failed with result 'exit-code
Mar 26 11:51:44 vm-150 systemd[1]: isso.service: Service hold-off time over, s
Mar 26 11:51:44 vm-150 systemd[1]: Stopped Isso Commenting Server.
Mar 26 11:51:44 vm-150 systemd[1]: Started Isso Commenting Server.
Mar 26 11:51:44 vm-150 isso[26482]: usage: isso [-h] [--version] [-c /etc/isso
Mar 26 11:51:44 vm-150 isso[26482]: isso: error: too few arguments
Mar 26 11:51:44 vm-150 systemd[1]: isso.service: Main process exited, code=exi
Mar 26 11:51:44 vm-150 systemd[1]: isso.service: Unit entered failed state.
Mar 26 11:51:44 vm-150 systemd[1]: isso.service: Failed with result 'exit-code
```

> sudo journalctl -xe

```
Mar 27 00:23:21 vm-150 systemd[1]: isso.service: Failed with result 'exit-code
Mar 27 00:23:26 vm-150 systemd[1]: isso.service: Service hold-off time over, s
Mar 27 00:23:26 vm-150 systemd[1]: Stopped Isso Commenting Server.
-- Subject: Unit isso.service has finished shutting down
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit isso.service has finished shutting down.
Mar 27 00:23:26 vm-150 systemd[1]: Started Isso Commenting Server.
-- Subject: Unit isso.service has finished start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit isso.service has finished starting up.
--
-- The start-up result is done.
Mar 27 00:23:26 vm-150 isso[13407]: usage: isso [-h] [--version] [-c /etc/isso
Mar 27 00:23:26 vm-150 isso[13407]: isso: error: too few arguments
Mar 27 00:23:26 vm-150 systemd[1]: isso.service: Main process exited, code=exi
Mar 27 00:23:26 vm-150 systemd[1]: isso.service: Unit entered failed state.
Mar 27 00:23:26 vm-150 systemd[1]: isso.service: Failed with result 'exit-code
Mar 27 00:23:29 vm-150 sudo[13409]: ghostuser : TTY=pts/0 ; PWD=/home/ghostuse
Mar 27 00:23:29 vm-150 sudo[13409]: pam_unix(sudo:session): session opened for
lines 2407-2429/2429 (END)

```

> /etc/systemd/system/isso.service

```
[Unit]
Description=Isso Commenting Server
After=network.target
[Service]
Type=simple
User=isso
WorkingDirectory=/home/isso
ExecStart=/home/isso/.local/bin/isso -c /home/isso/isso.conf
Restart=on-failure
RestartSec=5
[Install]
WantedBy=multi-user.target
```

> Permissions

`-rwxrw-r-- 1 isso isso 3468 Mar 24 20:29 isso.conf*`

> isso config

```
[general]
dbpath = /home/isso/comments.db
host = https://mydomain.com
max-age = 20m
notify = smtp
log-file = /home/isso/isso.log
admin_password = ADMINPASSWORD

[moderation]
enabled = false

[server]
listen = http://localhost:8080
reload = off
profile = off
public-endpoint = https://isso.mydomain.com

[guard]
enabled = true
ratelimit = 2
direct-reply = 3
reply-to-self = false
require-author = true
require-email = false

[smtp]
username =
password =
host =
port = 587
security = starttls
to =
from =
timeout = 10

[hash]
salt = 3EFech7co8Ohlsad4324aso6Adsbaimi
algorithm = pbkdf2
```

**Comment by rouja** [comment]
Hi,

Logs say : 
```
Mar 26 11:51:44 vm-150 isso[26482]: usage: isso [-h] [--version] [-c /etc/isso
Mar 26 11:51:44 vm-150 isso[26482]: isso: error: too few arguments
```

What's happen when you try to launch isso manually ?

```
sudo -i
su - isso -s /bin/bash
/home/isso/.local/bin/isso -c /home/isso/isso.conf
```

**Comment by inputuserhere** [comment]
It gives me the same error :/

```
usage: isso [-h] [--version] [-c /etc/isso.conf] {import,run} ...
isso: error: too few arguments
```

**Comment by rouja** [comment]
Hi,

It looks like there is a missing "run" argument.

Can you try this : 
```
sudo -i
su - isso -s /bin/bash
/home/isso/.local/bin/isso -c /home/isso/isso.conf run
```
If it's running, I think you need to fix your systemd service.

**Comment by inputuserhere** [comment]
Sorry for the late reply. I don't think it's my systemd, I believe it could be python, not sure if how Isso was built with pip.

> Traceback (most recent call last):
>   File "/home/isso/.local/bin/isso", line 8, in <module>
>     sys.exit(main())
>   File "/home/isso/.local/lib/python2.7/site-packages/isso/__init__.py", line 237, in main
>     join(dist.location, dist.project_name, "defaults.ini"), args.conf)
>   File "/home/isso/.local/lib/python2.7/site-packages/isso/config.py", line 136, in load
>     parser.read(user)
>   File "/home/isso/.local/lib/python2.7/site-packages/backports/configparser/__init__.py", line 770, in read
>     self._read(fp, filename)
>   File "/home/isso/.local/lib/python2.7/site-packages/backports/configparser/__init__.py", line 1148, in _read
>     raise DuplicateSectionError(sectname, fpname, lineno)
> backports.configparser.DuplicateSectionError: While reading from '/home/isso/isso.conf' [line 46]: section u'moderation' already exists

**Comment by blatinier** [comment]
You have two time the `moderation` in your isso.conf.
You should delete one.

**Comment by Konzertheld** [comment]
@inputuserhere Additional to what @blatinier said, you are missing the "run" at the end of the ExecStart command in your systemd configuration. It should look like this:

`ExecStart=/home/isso/.local/bin/isso -c /home/isso/isso.conf run`

**Comment by inputuserhere** [comment]
@Konzertheld 

I do have all the settings set under '[hash]' in my isso.conf file.
```
$ /home/isso/.local/bin/isso -c /home/isso/isso.conf run
2020-06-09 00:39:42,711 WARNING: no such option: [hash] host
2020-06-09 00:39:42,711 WARNING: no such option: [hash] max-age
2020-06-09 00:39:42,712 WARNING: no such option: [general] admin_password
2020-06-09 00:39:42,712 WARNING: no such option: [hash] notify
2020-06-09 00:39:42,712 WARNING: no such option: [hash] th
2020-06-09 00:39:42,712 WARNING: no such option: [hash] admin_password
2020-06-09 00:39:42,713 WARNING: no such option: [hash] log-file
Traceback (most recent call last):
  File "/home/isso/.local/bin/isso", line 8, in <module>
    sys.exit(main())
  File "/home/isso/.local/lib/python2.7/site-packages/isso/__init__.py", line 273, in main
    use_reloader=conf.getboolean('server', 'reload'))
  File "/home/isso/.local/lib/python2.7/site-packages/werkzeug/serving.py", line 1052, in run_simple
    inner()
  File "/home/isso/.local/lib/python2.7/site-packages/werkzeug/serving.py", line 1005, in inner
    fd=fd,
  File "/home/isso/.local/lib/python2.7/site-packages/werkzeug/serving.py", line 848, in make_server
    host, port, app, request_handler, passthrough_errors, ssl_context, fd=fd
  File "/home/isso/.local/lib/python2.7/site-packages/werkzeug/serving.py", line 740, in __init__
    HTTPServer.__init__(self, server_address, handler)
  File "/usr/lib/python2.7/SocketServer.py", line 417, in __init__
    self.server_bind()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 108, in server_bind
    SocketServer.TCPServer.server_bind(self)
  File "/usr/lib/python2.7/SocketServer.py", line 431, in server_bind
    self.socket.bind(self.server_address)
  File "/usr/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
socket.error: [Errno 98] Address already in use
```

systemctl status isso:

$ sudo systemctl status isso
```
● isso.service - Isso Commenting Server
   Loaded: loaded (/etc/systemd/system/isso.service; enabled; vendor preset: ena
   Active: activating (auto-restart) (Result: exit-code) since Tue 2020-06-09 00
  Process: 20144 ExecStart=/home/isso/.local/bin/isso -c /home/isso/isso.conf ru
 Main PID: 20144 (code=exited, status=1/FAILURE)

Jun 09 00:43:19 webserv systemd[1]: isso.service: Unit entered failed state.
Jun 09 00:43:19 webserv systemd[1]: isso.service: Failed with result 'exit-code
```


I have the below set in my isso.service
`ExecStart=/home/isso/.local/bin/isso -c /home/isso/isso.conf run`

**Comment by blatinier** [comment]
`socket.error: [Errno 98] Address already in use`
You already have a service running on the port you are trying to run isso on.
Change `listen = http://localhost:8080` into `listen = http://localhost:8081` in your config

**Comment by inputuserhere** [comment]
@blatinier Well I hope it's not due to my blog.. Before I moved servers it was working perfectly fine on port 8080, both services.

**Comment by blatinier** [comment]
That's highly improbable. Once a process listen on a port, others can't.
You can use `ss` to see what's running on this port:
`ss -tlp`
will show you TCP ports `t` listening sockets `l` and associated process `p`

**Comment by inputuserhere** [comment]
Everything's running correctly now. But the comment section doesn't show up on my pages, and yes I added the js portion to my site.

Does the isso domain need to be reached in order for it to show up? If so my isso.conf in /etc/nginx/conf.d

`server {
        listen 81;
        listen [::]:81;
        server_name isso.domain.com;
        return 301 https://isso.domain.com$request_uri;

        access_log  /dev/null;
        error_log /dev/null;
}
server {
        listen 443 ssl http2;
        listen [::]:443 ssl http2;
        server_name isso.domain.com;

        access_log /var/log/nginx/isso-access.log;
        error_log  /var/log/nginx/isso-error.log;

        ssl_certificate     /etc/nginx/https/fullchain.pem;
        ssl_certificate_key /etc/nginx/https/key.pem;


        location / {
                proxy_set_header Host $http_host;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_pass http://localhost:8081;
        }
}`

**Comment by blatinier** [comment]
`isso.domain.com` Needs to exists in the DNS records and it needs to point to your server running isso.

**Comment by inputuserhere** [comment]
It is created. But Isso refuses to show up on my site. Is does not need admin panel to work right, because  when I do systemctl status isso its fully operational.

**Comment by ix5** [comment]
What happens when you try to fetch `https://isso.domain.com/js/embed.min.js`?

Also, is this still a current issue of yours?

**Comment by ix5** [comment]
@inputuserhere could you kindly verify this is still an issue?
